File: problem804/problem804.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        // ユニーク数だけ知りたいので集合に追加していく
        set<string> res;
        string mc[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};

        // wordsに含まれる単語それぞれをモールス信号に変換
        for (const auto& word : words) {
            string mc_word = "";
            for(int j = 0; j<word.length(); j++) mc_word += mc[word[j] - 'a'];
            res.insert(mc_word);
        }
        return res.size();
    }
};

File: problem804/problem804.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        mmap = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        res = []
        oda, odz = ord("a"), ord("z")

        for word in words:
            mc_word = ""
            for c in word:
                od = ord(c)
                mc_word += mmap[od-97]
            res.append(mc_word)

        return len(set(res))